library/general/test/cfa/nsswitch_test.rb

```diff
@@ -0,0 +1,190 @@
+# Copyright (c) [2020] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require_relative "../test_helper"
+require "cfa/nsswitch"
+
+describe CFA::Nsswitch do
+  subject { described_class.new(file_handler: file_handler) }
+
+  let(:file_handler) { Yast::TargetFile }
+
+  let(:scenario) { "custom" }
+
+  around do |example|
+    change_scr_root(File.join(GENERAL_DATA_PATH, "nsswitch.conf", scenario), &example)
+  end
+
+  describe "#load" do
+    context "when /etc/nsswitch.conf exists" do
+      before do
+        allow(file_handler).to receive(:read).and_call_original
+      end
+
+      it "does not read /usr/etc/nsswitch.conf file" do
+        expect(file_handler).to_not receive(:read)
+          .with("/usr/etc/nsswitch.conf")
+        subject.load
+      end
+
+      it "reads /etc/nsswitch.conf file" do
+        expect(file_handler).to receive(:read)
+          .with("/etc/nsswitch.conf")
+          .and_call_original
+        subject.load
+      end
+    end
+
+    context "when /etc/nsswitch.conf does not exist" do
+      let(:scenario) { "vendor" }
+
+      it "reads vendor files" do
+        expect(file_handler).to receive(:read)
+          .with("/usr/etc/nsswitch.conf")
+          .and_call_original
+        subject.load
+      end
+    end
+  end
+
+  describe "#save" do
+    before do
+      subject.load
+
+      # Avoid writing data example files
+      allow(file_handler).to receive(:write)
+    end
+
+    context "when it has changed" do
+      before do
+        subject.update_entry("hosts", ["dns", "nis"])
+      end
+
+      it "returns true" do
+        expect(subject.save).to eq(true)
+      end
+
+      it "writes to /etc/nsswitch.conf file" do
+        expect(file_handler).to receive(:write)
+          .with("/etc/nsswitch.conf", anything)
+        subject.save
+      end
+
+      it "does not write to /usr/etc/nsswitch.conf file" do
+        expect(file_handler).to_not receive(:write)
+          .with("/user/etc/nsswitch.conf", anything)
+        subject.save
+      end
+    end
+
+    context "when it has not changed" do
+      it "returns false" do
+        expect(subject.save).to eq(false)
+      end
+
+      it "does nothing" do
+        expect(file_handler).to_not receive(:write)
+        subject.save
+      end
+    end
+  end
```

Comment:
as we quite depends on lense that is not under our control, but we need to be sure we use it correct for both save/load I prefer to do real file read/write and verify that it is what we expect. file handler MemoryFile is your friend. See https://github.com/config-files-api/config_files_api/blob/master/spec/augeas_writer_spec.rb

Reply:
I'll take a look, thanks!